File: api/Ez.Hress.UserProfile/Entities/BalanceSheet.cs
namespace Ez.Hress.UserProfile.Entities;

public class BalanceSheet
{
    public BalanceSheet()
    {
        Transactions = new List<Transaction>();
    }

    public int UserID { get; set; }
    public int Balance { get; set; }

    public IList<Transaction> Transactions { get; set; }
}

File: src/types/balanceSheet.ts
import { UserBasicEntity } from "./userBasicEntity";

export interface Transaction {
  user: UserBasicEntity;
  amount: number;
  id: number;
  name: string;
  inserted: string;
  insertedString: string;
  insertedBy: number;
  deleted: string;
}

export interface BalanceSheet {
  userID: number;
  balance: number;
  transactions: Transaction[];
}
